docs/search.md

```diff
@@ -0,0 +1,199 @@
+
+The `Document` class includes three things:
+
+1. The [Elasticsearch index configuration](https://django-elasticsearch-dsl.readthedocs.io/en/latest/quickstart.html#declare-data-to-index)
+2. The Django model information
+3. [Custom fields to index](https://django-elasticsearch-dsl.readthedocs.io/en/latest/fields.html), and any preparation that they require
+
+We'll use our [Ask CFPB answer search document as an example](https://github.com/cfpb/consumerfinance.gov/blob/main/cfgov/ask_cfpb/documents.py) for each of these:
+
+```python
+from django_elasticsearch_dsl import Document
+from django_elasticsearch_dsl.registries import registry
+
+@registry.register_document
+class AnswerPageDocument(Document):
+    pass
+```
+
+### Elasticsearch index configuration
+
+The index configuration is provided by an `Index` class on the document class that defines the [elasticsearch-dsl index options](https://elasticsearch-dsl.readthedocs.io/en/latest/persistence.html#class-index-options).
+
+```python
+from search.elasticsearch_helpers import environment_specific_index
+
+class AnswerPageDocument(Document):
+
+    class Index:
+        name = environment_specific_index('ask-cfpb')
+        settings = {'number_of_shards': 1,
+                    'number_of_replicas': 0}
+```
+
+Fot index naming, we have a helper function, `environment_specific_index`, that will generate the index name specific to the deployment environment. This allows each index to be isolated to a deployment environment within an Elasticsearch cluster.
+
+### Django model information
+
+The Django model information is provided by a `Django` class on the document class that defines the model and any fields names to be indexed directly from the model. A `get_queryset` method can be overriden to perform any filtering on the model's queryset before content is indexed.
+
+```python
+from ask_cfpb.models.answer_page import AnswerPage
+
+class AnswerPageDocument(Document):
+
+    def get_queryset(self):
+        query_set = super().get_queryset()
+        return query_set.filter(live=True, redirect_to_page=None)
+
+    class Django:
+        model = AnswerPage
+
+        fields = [
+            'search_tags',
+            'language',
+        ]
+```
+
+The fields in `fields` will be indexed without any preparation/manipulation, directly as are stored on the model. 
+
+### Custom fields
+
+Sometimes it might be desirable to index a field as an alternative type — say, the string that matches an integer for a Django field that species `choices`. 
+
+It might also be desirable to construct a field to index from multiple fields on the model, particular for Wagtail pages with stream fields. 
```

Comment:
```suggestion
It might also be desirable to construct a field to index from multiple fields on the model, particularly for Wagtail pages with stream fields. 
```